### DISCUSSION TITLE
Question About Support Dates for OpenSSL 3.3.1 and FIPS 3.0.9

### DISCUSSION THREAD

**Discussion by mahenderkk08** [openssl/openssl#25480] [Q&A] [answered]
Hi Team,

I’m using OpenSSL 3.3.1 and FIPS 3.0.9, and I want to make sure I have the right information about how long these versions will be supported. According to the OpenSSL website:

The 3.3 series (including 3.3.1) is supported until April 9, 2026.?
The 3.0 series, which is a Long Term Support (LTS) version, is supported until September 7, 2026.
Could you please confirm if this is correct? Also, does this support timeline apply to the FIPS 3.0.9 module as well?

Thanks a lot for your help!

Regards,
Mahender K

**Comment by mattcaswell** [comment]
> The 3.3 series (including 3.3.1) is supported until April 9, 2026.?

Yes

> The 3.0 series, which is a Long Term Support (LTS) version, is supported until September 7, 2026.

Yes

> Also, does this support timeline apply to the FIPS 3.0.9 module as well?

Yes

**Reply by mahenderkk08** [reply]
thanks for the confirmation @mattcaswell

**Comment by tom-cosgrove-arm** [comment]
@mattcaswell Would I be right in thinking that 3.3.1 isn't specifically supported until April 2026: we now have 3.3.2, and bug fixes for the 3.3 series will continue in 3.3.3, 3.3.4, etc. That is, there won't be a 3.3.1a, 3.3.1b, 3.3.1c, etc?

**Reply by mattcaswell** [reply]
Correct

**Comment by mahenderkk08** [comment]
Hi @mattcaswell @t8m  , could you please confirm whether CVE-2024-5535 affects OpenSSL versions **3.3.1**, **FIPS 3.0.9**, and **1.0.2zj** ? 

Additionally, could you provide information on the available fixed versions? 

Thank you!

**Comment by mattcaswell** [comment, marked as answer]
You can find information about specific vulnerabilities on our vulnerabilities page. For that particular CVE see here:

https://openssl-library.org/news/vulnerabilities/#CVE-2024-5535

On that page you can see this text:

> Fixed in OpenSSL 3.3.2 [(git commit)](https://github.com/openssl/openssl/commit/e86ac436f0bd54d4517745483e2315650fae7b2c) (Affected since 3.3.0)
> Fixed in OpenSSL 3.2.3 [(git commit)](https://github.com/openssl/openssl/commit/99fb785a5f85315b95288921a321a935ea29a51e) (Affected since 3.2.0)
> Fixed in OpenSSL 3.1.7 [(git commit)](https://github.com/openssl/openssl/commit/4ada436a1946cbb24db5ab4ca082b69c1bc10f37) (Affected since 3.1.0)
> Fixed in OpenSSL 3.0.15 [(git commit)](https://github.com/openssl/openssl/commit/cf6f91f6121f4db167405db2f0de410a456f260c) (Affected since 3.0.0)
> Fixed in OpenSSL 1.1.1za [(premium support)](https://openssl-corporation.org/support/?giturl=https://github.openssl.org/openssl/extended-releases/commit/b78ec0824da857223486660177d3b1f255c65d87%22) (Affected since 1.1.1)
> Fixed in OpenSSL 1.0.2zk [(premium support)](https://openssl-corporation.org/support/?giturl=https://github.openssl.org/openssl/extended-releases/commit/9947251413065a05189a63c9b7a6c1d4e224c21c%22) (Affected since 1.0.2)

So, 3.3.1 is affected and the issue is fixed in 3.3.2. Also 1.0.zj is affected and it is fixed in 1.0.2zk.

There is also the following text:

> The FIPS modules in 3.3, 3.2, 3.1 and 3.0 are not affected by this issue.

So the FIPS 3.0.9 module is unaffected. There is also this page that summarises whether a CVE impacts the FIPS module or not:

https://openssl-library.org/news/fips-cve/index.html

**Reply by mahenderkk08** [reply]
Hi @mattcaswell  @t8m  Thank you for the clarification!

From the details you provided, I understand that:

- OpenSSL version 3.3.1 is affected by CVE-2024-5535, and the issue is fixed in version 3.3.2. However, we are considering upgrading to the **latest version, 3.4.0.** Could you confirm if all the fixes for CVE-2024-5535 are included in this version? I wasn’t able to find detailed information on this.

- OpenSSL 1.0.2zj is also affected, with the fix available in version 1.0.2zk.

- The FIPS 3.0.9 module is unaffected by this issue.

Thanks again for your help!

**Comment by mahenderkk08** [comment]
Hi @t8m @mattcaswell , Could you kindly confirm if the fix for CVE-2024-5535 is included in the latest OpenSSL version 3.4.0?

Thank you in advance for your help!

**Reply by t8m** [reply]
As 3.4.0 release was released after 3.3.2, it is of course not affected by CVE-2024-5535
